packages/react-router-dom/index.tsx

```diff
@@ -1318,11 +1318,15 @@ function useScrollRestoration({
         let key = (getKey ? getKey(location, matches) : null) || location.key;
         savedScrollPositions[key] = window.scrollY;
       }
-      sessionStorage.setItem(
-        storageKey || SCROLL_RESTORATION_STORAGE_KEY,
-        JSON.stringify(savedScrollPositions)
-      );
-      window.history.scrollRestoration = "auto";
+      try {
+        sessionStorage.setItem(
+          storageKey || SCROLL_RESTORATION_STORAGE_KEY,
+          JSON.stringify(savedScrollPositions)
+        );
+        window.history.scrollRestoration = "auto";
```

Comment:
We should probably keep this outside the `try` so it gets properly reset even if the `sessonStorage` set fails